plugins/catalog-backend/src/ingestion/processors/ldap/client.ts

```diff
@@ -18,12 +18,24 @@ import ldap, { Client, SearchEntry, SearchOptions } from 'ldapjs';
 import { BindConfig } from './config';
 import { errorString } from './util';
 
+/**
+ * A string representation of a Microsoft Active Directory Server.
+ */
+export const LDAP_VENDOR_ACTIVE_DIRECTORY = 'Microsoft Active Directory';
+
+/**
+ * A default string representation of an unknown LDAP server vendor.
+ */
+export const LDAP_VENDOR_UNKNOWN = 'Unknown';
+
 /**
  * Basic wrapper for the ldapjs library.
  *
  * Helps out with promisifying calls, paging, binding etc.
  */
 export class LdapClient {
+  private vendor = '';
```

Comment:
With a thing like this, it's often nice to hold on to a promise that other things chain off of.

```ts
private vendor: Promise<string> | undefined;

public async doSomething(): Promise<something> {
  const vendor = await this.getVendor();
  // use the vendor
}

private getVendor(): Promise<string> {
  if (this.vendor) {
    // somebody is already either in the process of computing this, or has
    // successfully computed it
    return this.vendor;
  }

  this.vendor = this.computeVendor().catch(err => {
    // oh we failed; now we'll make sure to try again at a later time,
    // and also rethrow to everybody that was waiting for the result
    this.vendor = undefined;
    throw e;
  });
}

private async computeVendor(): Promise<string> {
  // do what's necessary
}
```

Reply:
Thanks. I have updated my implementation to follow your suggestion